src/latexify/codegen/algorithm_codegen.py

```diff
@@ -0,0 +1,67 @@
+import ast
+
+from latexify import exceptions
+from latexify.codegen import function_codegen
+
+
+class AlgorithmCodegen(function_codegen.FunctionCodegen):
```

Comment:
Just a quick comment: This class shouldn't inherit `FunctionCodegen` to avoid a weird "is-a" class relationship, e.g., `isinstance(AlgorithmCodegen(), FunctionCodegen)` should be `False`. Inheritance also brings all functionality in `FunctionCodegen` implicitly, which may increase the maintenance cost in the future.

`FunctionCodegen` can be instantiated as a private member of this class (aka composition) and used internally only when we need to delegate the codegen logic to it.